flink-test-utils-parent/flink-connector-testing-framework/src/main/java/org/apache/flink/connectors/test/common/junit/extensions/TestingFrameworkExtension.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connectors.test.common.junit.extensions;
+
+import org.apache.flink.connectors.test.common.TestResource;
+import org.apache.flink.connectors.test.common.environment.TestEnvironment;
+import org.apache.flink.connectors.test.common.external.ExternalContext;
+import org.apache.flink.connectors.test.common.external.ExternalSystem;
+import org.apache.flink.connectors.test.common.junit.annotations.WithExternalContextFactory;
+import org.apache.flink.connectors.test.common.junit.annotations.WithExternalSystem;
+import org.apache.flink.connectors.test.common.junit.annotations.WithTestEnvironment;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+import java.lang.annotation.Annotation;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * A JUnit 5 {@link Extension} for supporting running of testing framework.
+ *
+ * <p>This extension is responsible for searching test resources annotated by {@link
+ * WithTestEnvironment}, {@link WithExternalSystem} and {@link WithExternalContextFactory}, storing
+ * them into storage provided by JUnit, and manage lifecycle of these resources.
+ *
+ * <p>The extension uses {@link ExtensionContext.Store} for handing over test resources to {@link
+ * TestCaseInvocationContextProvider}, which will inject these resources into test cases.
+ */
+public class TestingFrameworkExtension implements BeforeAllCallback, AfterAllCallback {
```

Comment:
Are you expecting usage beyond connectors? If not, we could name it `ConnectorTestingExtension` (again dropping framework). If it's general, then we should find a new package/module name.

Reply:
Renamed to ```ConnectorTestingExtension```. I'd like to use it just for connector testing framework because this extension needs to work together with ```TestCaseInvocationContextProvider```